src/openrct2/actions/StaffHireNewAction.hpp

```diff
@@ -0,0 +1,345 @@
+        }
+
+        if (execute == false)
+        {
+            // In query we just want to see if we can obtain a sprite slot.
+            sprite_remove((rct_sprite*)newPeep);
+        }
+        else
+        {
+            move_sprite_to_list((rct_sprite*)newPeep, SPRITE_LIST_PEEP * 2);
+
+            newPeep->sprite_identifier = 1;
+            newPeep->window_invalidate_flags = 0;
+            newPeep->action = PEEP_ACTION_NONE_2;
+            newPeep->special_sprite = 0;
+            newPeep->action_sprite_image_offset = 0;
+            newPeep->no_action_frame_num = 0;
+            newPeep->action_sprite_type = PEEP_ACTION_SPRITE_TYPE_NONE;
+            newPeep->path_check_optimisation = 0;
+            newPeep->type = PEEP_TYPE_STAFF;
+            newPeep->outside_of_park = 0;
+            newPeep->peep_flags = 0;
+            newPeep->paid_to_enter = 0;
+            newPeep->paid_on_rides = 0;
+            newPeep->paid_on_food = 0;
+            newPeep->paid_on_souvenirs = 0;
+
+            if (_staffType == STAFF_TYPE_HANDYMAN)
+                newPeep->staff_orders = STAFF_ORDERS_SWEEPING | STAFF_ORDERS_WATER_FLOWERS | STAFF_ORDERS_EMPTY_BINS;
+            else if (_staffType == STAFF_TYPE_MECHANIC)
+                newPeep->staff_orders = STAFF_ORDERS_INSPECT_RIDES | STAFF_ORDERS_FIX_RIDES;
+            else
+                newPeep->staff_orders = 0;
+
+            uint16_t idSearchSpriteIndex;
+            Peep* idSearchPeep;
+
+            // We search for the first available id for a given staff type
+            uint32_t newStaffId = 0;
+            for (;;)
+            {
+                bool found = false;
+                ++newStaffId;
+
+                FOR_ALL_STAFF (idSearchSpriteIndex, idSearchPeep)
+                {
+                    if (idSearchPeep->staff_type != _staffType)
+                        continue;
+
+                    if (idSearchPeep->id == newStaffId)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                    break;
+            }
+
+            newPeep->id = newStaffId;
+            newPeep->staff_type = _staffType;
+
+            PeepSpriteType spriteType = spriteTypes[_staffType];
+            if (_staffType == STAFF_TYPE_ENTERTAINER)
+            {
+                spriteType = static_cast<PeepSpriteType>(PEEP_SPRITE_TYPE_ENTERTAINER_PANDA + _entertainerType);
+            }
+            newPeep->name_string_idx = staffNames[_staffType];
+            newPeep->sprite_type = spriteType;
+
+            const rct_sprite_bounds* spriteBounds = g_peep_animation_entries[spriteType].sprite_bounds;
+            newPeep->sprite_width = spriteBounds->sprite_width;
+            newPeep->sprite_height_negative = spriteBounds->sprite_height_negative;
+            newPeep->sprite_height_positive = spriteBounds->sprite_height_positive;
+
+            if (_autoPosition == true)
```

Comment:
Shouldn't this be passed into the function instead of a global

Reply:
_autoPosition is a member not a global.